src/rgw/rgw_basic_types.h

```diff
@@ -68,9 +82,19 @@ struct rgw_user {
     size_t pos = str.find('$');
     if (pos != std::string::npos) {
       tenant = str.substr(0, pos);
-      id = str.substr(pos + 1);
+      string_view sv = str;
+      string_view ns_id = sv.substr(pos + 1);
+      size_t ns_pos = ns_id.find('$');
```

Comment:
won't reuse of `$` as a delimiter for namespace cause parsing ambiguities? won't `user$oidc` get parsed as tenant=user uid=oidc instead?

Reply:
please suggest an alternative so pritha can send another change

Reply:
@cbodley

Reply:
I didn't notice the delimiter at all, and maybe @adamemerson didn't either, so making this more salient in the code--e.g., through use of a constant--would be helpful; did we also miss parsing unit tests?

Reply:
@mattbenjamin @cbodley @adamemerson : The users will be saved as tenant$oidc$uid and while parsing this all three fields will be properly parsed out. If the tenant is "" or the global tenant then the user will be saved as $oidc$uid, so in this case also all three fields will be properly parsed. For other users which are of the format tenant$uid, the logic will parse out the tenant and uid properly. Is this ok, or is there still a case that has not been taken into account?

Reply:
> If the tenant is "" or the global tenant then the user will be saved as $oidc$uid, so in this case also all three fields will be properly parsed

okay, that's clever! that should be fine then

Reply:
that's good to hear;  what I'd add is, I do think we want to make this (parsing, parsing rules, &c) clearer in the code and maybe doc